src/app/pages/videos/videos-list/shared/scroll-tracker.directive.ts

```diff
@@ -0,0 +1,20 @@
+import { Directive, Output, EventEmitter, HostListener } from '@angular/core';
+
+@Directive({
+  selector: '[scrollTracker]'
+})
+export class ScrollTrackerDirective {
+  @Output() scrollingFinished = new EventEmitter<void>();
+
+  emitted = false;
+
+  @HostListener("window:scroll", [])
+  onScroll(): void {
+    if ((window.innerHeight + window.scrollY) >= document.body.offsetHeight && !this.emitted) {
+      this.emitted = true;
+      this.scrollingFinished.emit();
+    } else if ((window.innerHeight + window.scrollY) < document.body.offsetHeight) {
+      this.emitted = false;
+    }
+  }
```

Comment:
Can you please write comments on this to understand how it works. 

Can we use a debounce on this. If not that is okay

Reply:
In this I am trying to find the height of the scrolling content if it reached at the end of the page, I am trying to emit so that I can caputure and make an API call.

If the scrolling content is in the top or in middle I am not emitting.

Reply:
I will try using debounce and update you